src/components/__tests__/button.test.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2018 Ultimaker B.V.
+import 'jest';
+import 'jsdom-global/register';
+import * as React from 'react';
+import { shallow } from 'enzyme';
+
+// component
+import Button from '../button';
+import Spinner from '../spinner';
+
+// mocks
+import { mockClickEvent } from '../../__mocks__/clickMock'
+
+describe('The Button component', () => {
+  let props;
+  let wrapper;
+
+  beforeEach(() => {
+    props = {
+      onClickHandler: jest.fn(),
+      id: 'testButton'
+    };
+    wrapper = shallow(<Button {...props}>Button text</Button>);
+  });
+
+  it('should render', () => {
+    expect(wrapper).toMatchSnapshot();
+    expect(props.onClickHandler).not.toHaveBeenCalled();
+  });
+
+  it('calls onClickHandler when it is clicked', () => {
+    wrapper.simulate('click', mockClickEvent);
+    expect(props.onClickHandler).toHaveBeenCalled();
+  });
+
+  it('does not call onClickHandler when it is clicked and onClickHandler is not passed', () => {
+    wrapper.setProps({ onClickHandler: null });
+    wrapper.simulate('click', mockClickEvent);
+    expect(props.onClickHandler).not.toHaveBeenCalled();
+  });
+
+  it('renders a button with type submit prop `type` is submit', () => {
+    wrapper.setProps({ type: 'submit' });
+    expect(wrapper.find('[type="submit"]')).toHaveLength(1);
+  });
+
+  it('renders a anchor when prop `type` is link', () => {
+    wrapper.setProps({ type: 'link', linkURL: 'https://ultimaker.com/' });
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('is disabled when prop `disabled` is true', () => {
```

Comment:
Maybe it's worth testing that it does not call the onClickHandler when the button is disabled?